plane/src/client/mod.rs

```diff
@@ -70,11 +70,17 @@ impl PlaneClient {
     pub fn drone_connection(
         &self,
         cluster: &ClusterName,
+        pool: Option<String>,
     ) -> TypedSocketConnector<MessageFromDrone> {
-        let addr = self
-            .controller_address
-            .join(&format!("/ctrl/c/{}/drone-socket", cluster))
-            .to_websocket_address();
+        let base_path = format!("/ctrl/c/{}/drone-socket", cluster);
+        let addr = match pool {
+            Some(pool_name) => self
+                .controller_address
+                .join(&format!("{}?pool={}", base_path, pool_name)),
```

Comment:
We should encode `pool_name` here (no need to decode it on the other end; Axum will already do that since it's a query param).

I'm also wondering if we should make it part of the path, though -- maybe `/ctrl/c/{}/drone-socket` is the default (empty string) pool, and then `/ctrl/c/{}/p/{}/drone-socket` is the one for a pool.

The main factor for me in preferring it in the path is that when run behind a reverse proxy, it is generally easier to create path-based rules, so a user could create a rule like "if a drone tries to connect as part of node pool 'abc', require the bearer token 'xyz'". When we've had requests for pools, it was often from a desire to create trust barriers between drones within a cluster, so this would provide a good way to ensure that drones from one pool couldn't "impersonate" drones from another even if entirely compromised.

Reply:
I had a fun deep dive on how Axum's extractors work to make this happen in one handler, curious what you think of the result.

The less fancy and more explicit way to do it is just having two separate handlers. With the explicit error handling I've done, I believe the difference in the two approaches is simply style preference.

Reply:
After some offline discussion (see https://github.com/drifting-in-space/plane/pull/614#discussion_r1502970060) we are moving back to using query params (instead of the path approach)